pandas/core/reshape/pivot.py

```diff
@@ -493,6 +494,13 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     rownames = _get_names(index, rownames, prefix='row')
     colnames = _get_names(columns, colnames, prefix='col')
 
+    obs_idxes = [obj.index for objs in (index, columns) for obj in objs
```

Comment:
Not a fan of this solution (the list comprehension would look only slightly better). What about adding a new method ``_get_objs_combined_index`` which gets a list of objects (with or without index) and does the above?